cfme/services/requests.py

```diff
@@ -1,288 +1,498 @@
+            cancel: Whether to cancel the denial.
+        """
+        self.rest.action.deny(reason=reason)
+
+    @deny_request.variant('ui')
+    def deny_request_ui(self, reason, cancel=False):
+        view = navigate_to(self, 'Deny')
+        view.reason.fill(reason)
+        if not cancel:
+            view.submit.click(handle_alert=not cancel)
+        else:
+            view.breadcrumb.click_location(view.breadcrumb.locations[0], handle_alert=True)
+        view.flash.assert_no_error()
+
+    def remove_request(self, cancel=False):
+        """Opens the specified request and deletes it - removes from UI
+        Args:
+            cancel: Whether to cancel the deletion.
+        """
+        view = navigate_to(self, 'Details')
+        view.toolbar.delete.click(cancel)
+
+    @variable(alias='rest')
+    def is_finished(self):
+        """Helper function checks if a request is completed
+        """
+        self.update()
+        return self.rest.request_state.title() in self.REQUEST_FINISHED_STATES
+
+    @is_finished.variant('ui')
+    def is_finished_ui(self):
+        self.update(method='ui')
+        return self.row.request_state.text in self.REQUEST_FINISHED_STATES
+
+    @variable(alias='rest')
+    def is_succeeded(self):
+        return self.is_finished() and self.rest.status.title() == 'Ok'
+
+    @is_succeeded.variant('ui')
+    def is_succeeded_ui(self):
+        return self.is_finished(method=('ui')) and self.row.status.text == 'Ok'
+
+    def copy_request(self, values=None):
+        """Copies the request  and edits if needed
+        """
+        view = navigate_to(self, 'Copy')
+        if view.form.fill(values):
+            view.submit_button.click()
+        else:
+            logger.debug('Making copy of current request without any changes')
+            view.submit_button.click()
+        # The way we identify request is a description which is based on vm_name,
+        # no need returning Request obj if name is the same => raw request copy
+        if 'vm_name' in values.keys():
+            return Request(description=values['vm_name'], partial_check=True)
+
+    def edit_request(self, values):
+        """Opens the request for editing and saves or cancels depending on success.
+        """
+        view = navigate_to(self, 'Edit')
+        if view.form.fill(values):
+            view.form.submit_button.click()
+            self.update()
+        else:
+            logger.debug('Nothing was changed in current request')
+
+
+class RequestBasicView(BaseLoggedInPage):
+    title = Text('//div[@id="main-content"]//h1')
+
+    @property
+    def in_requests(self):
+        return self.logged_in_as_current_user
+        # TODO uncomment after BZ 1472844 is fixed
+        # and  self.navigation.currently_selected == ['Services', 'Requests'] and
+
+
+class RequestsView(RequestBasicView):
+    table = Table(locator='//*[@id="list_grid"]/table')
+    pagination_pane = PaginationPane()
```

Comment:
I kept using old paginator due to BZ in find_request - this was left for future replacement